File: 0053-maximum-subarray/0053-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int i=0,sum=nums[0],m=nums[0];
        for(i=1;i<nums.size();i++)
        {
            sum=max(nums[i],sum+nums[i]);
            m=max(m,sum);
        }
        return m;
    }
};

File: 0053-maximum-subarray/0053-maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        msum,s=nums[0],nums[0]
        for i in range(1,len(nums)):
            s=max(nums[i],s+nums[i])
            msum=max(msum,s)
        return msum
        